crates/telnet-host/tests/common/mod.rs

```diff
@@ -123,70 +89,17 @@ fn start_telnet_host() -> ManagedChild {
     )
 }
 
-pub struct Client {
-    stream: TcpStream,
-}
-impl Client {
-    fn new(port: u16) -> Result<Self, std::io::Error> {
-        TcpStream::connect(format!("localhost:{port}")).and_then(|stream| {
-            stream.set_read_timeout(Some(Duration::from_secs(1)))?;
-            Ok(Self { stream })
-        })
-    }
-
-    fn send_string<S>(&mut self, s: S) -> Result<(), std::io::Error>
-    where
-        S: AsRef<str>,
-    {
-        eprintln!(">> {}", s.as_ref());
-        self.stream.write_all(s.as_ref().as_bytes())?;
-        self.stream.write_all(b"\n")?;
-        self.stream.flush()
-    }
-
-    pub fn command<S>(&mut self, s: S) -> Result<String, std::io::Error>
-    where
-        S: AsRef<str>,
-    {
-        self.send_string(s)?;
-
-        let mut lines = Vec::new();
-        let mut reader = BufReader::new(&self.stream);
-
-        // Wait for prefix
-        loop {
-            let mut line = String::new();
-            reader.read_line(&mut line)?;
-            if line == "-=!-^-!=-\n" {
-                break;
-            }
-        }
-
-        // Read until suffix
-        loop {
-            let mut line = String::new();
-            reader.read_line(&mut line)?;
-            if line == "-=!-v-!=-\n" {
-                break;
-            }
-            eprintln!("<< {}", line.strip_suffix('\n').unwrap());
-            lines.push(line);
-        }
-        Ok(lines.join(""))
-    }
-}
-
 pub fn run_test_as<F>(connect_params: &[&str], f: F) -> eyre::Result<()>
 where
-    F: FnOnce(Client) -> eyre::Result<()>,
+    F: FnOnce(MootClient) -> eyre::Result<()>,
 {
     let daemon_workdir = tempfile::TempDir::new()?;
     let _daemon = start_daemon(daemon_workdir.path());
     let _telnet_host = start_telnet_host();
 
     let start = Instant::now();
     loop {
-        if let Ok(mut client) = Client::new(8080) {
+        if let Ok(mut client) = MootClient::new(8080) {
```

Comment:
Could probably be done in a later submission, but we should probably use a dynamically allocated port somehow for this and for MOOT_PORT above, to keep the tests hermetic and isolated.

There's likely some crate out there that does this though I wasn't able to find one with a quick search.

Reply:
Hm. Each test starts its own daemon + telnet host, so there *should* be no way for this to go wrong. I think.

Reply:
It becomes an issue if you have the same port used by other things on the same container, no?

Or if you have a daemon running listening on the same ports, etc. I've seen some fails recently when running tests while having moor-daemon running locally (no container).